src/nisarqa/input_product_readers/nisar_product.py

```diff
@@ -596,6 +597,65 @@ def science_freq(self) -> str:
         """
         return "A" if "A" in self.freqs else "B"
 
+    def center_freq(self, freq: str) -> float:
+        """
+        The center frequency for input product's Frequency `freq`.
+
+        Parameters
+        ----------
+        freq : str
+            Must be either "A" or "B".
+
+        Returns
+        -------
+        center_freq : float
+            The center frequency for input product's Frequency `freq`.
+        """
+        center_freq_path = f"{self.get_freq_path(freq=freq)}/centerFrequency"
```

Comment:
This doesn't work for RSLC. It has "processedCenterFrequency" and "acquiredCenterFrequency".

https://github-fn.jpl.nasa.gov/NISAR-ADT/NISAR_PIX/blob/7b59f5f09d770952a277135b8130aac536786da9/XML/L1/nisar_L1_RSLC.xml#L294-L307

Let's move it to the InSAR product reader, rather than this base class that's common to all NISAR products.